File: C/Arreglos/Sort/ARR_ORD_3.c
//Insertion Sort

#include <stdio.h>
#include <stdlib.h> 
#include <time.h> 


void CARGA_VECTOR(int *v) 
{
	int i;

	//printf("%p\n", v);

	for(i = 0; i < 10; ++i) {		
		//Genero numeros entre 0 y 49
		v[i] = rand() % 50;
		}
}


void ORDENAR_VECTOR(int *v) 
{
	int k, aux, j;  

    for (k = 1; k < 10; k++) 
    {  
        aux = v[k];  
        j = k - 1;  
  
        while (j >= 0 && v[j] > aux) 
        {  
            v[j + 1] = v[j];  
            j = j - 1;  
        }  
        
        v[j + 1] = aux;  
    }  
}


int main() 
{	
	int i;
	int V[10];

	// Uso el tiempo actual para el generador de numeros 
    srand(time(0)); 

	CARGA_VECTOR(V);

	printf(" ---------------------\n");
	printf(" ----- RESULTADOS ----\n");
	printf(" ---------------------\n");

	printf("Vector origen: \n");

	//valores de los elementos del vector
	for(i = 0; i < 10; ++i) {
		printf("%d ", V[i]);
	}

	printf("\n ---------------------\n");

	ORDENAR_VECTOR(V);

	printf("Vector ordenado: \n");

	//valores de los elementos del vector
	for(i = 0; i < 10; ++i) {
		printf("%d ", V[i]);
	}

	return 0;
}

File: Python/Arreglos/Sort/ARR_ORD_3.py
import random

def CARGA_VECTOR(v):
    for i in range(len(v)):
        v[i] = random.randint(0, 49)

def ORDENAR_VECTOR(v):
    for k in range(1, len(v)):
        aux = v[k]
        j = k - 1
        while j >= 0 and v[j] > aux:
            v[j + 1] = v[j]
            j -= 1
        v[j + 1] = aux

def main():
    V = [0]*10
    random.seed()
    CARGA_VECTOR(V)
    print(" ---------------------")
    print(" ----- RESULTADOS ----")
    print(" ---------------------")
    print("Vector origen: ")
    for val in V:
        print(val, end=' ')
    print()
    print(" ---------------------")
    ORDENAR_VECTOR(V)
    print("Vector ordenado: ")
    for val in V:
        print(val, end=' ')
    print()

if __name__ == "__main__":
    main()